src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.Tracing.cs

```diff
@@ -124,5 +127,91 @@ public async Task SendAnInvocationMessage_SendTraceHeaders()
             }
         }
 
+        [Fact]
+        public async Task InvokeSendsAnInvocationMessage_ConnectionRemoteEndPointChanged_UseRemoteEndpointUrl()
+        {
+            var clientSourceContainer = new SignalRClientActivitySource();
+            Activity clientActivity = null;
+
+            using var listener = new ActivityListener
+            {
+                ShouldListenTo = activitySource => ReferenceEquals(activitySource, clientSourceContainer.ActivitySource),
+                Sample = (ref ActivityCreationOptions<ActivityContext> _) => ActivitySamplingResult.AllData,
+                ActivityStarted = activity => clientActivity = activity
+            };
+            ActivitySource.AddActivityListener(listener);
+
+            TestConnection connection = null;
+            connection = new TestConnection(onStart: () =>
+            {
+                connection.RemoteEndPoint = new UriEndPoint(new Uri("http://example.net"));
```

Comment:
nit: Should this be `"http://example.net:5050"` and then asserting `"server.port"`?